SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestCharacterMigration/SpatialTestCharacterMigration.cpp

```diff
@@ -76,89 +86,84 @@ void ASpatialTestCharacterMigration::PrepareTest()
 	// Move character forward
 	FSpatialFunctionalTestStepDefinition MoveForwardStepDefinition(/*bIsNativeDefinition*/ true);
 	MoveForwardStepDefinition.StepName = TEXT("Client1MoveForward");
-	MoveForwardStepDefinition.TimeLimit = 0.0f;
+	MoveForwardStepDefinition.TimeLimit = 5.0f;
 	MoveForwardStepDefinition.NativeTickEvent.BindLambda([this](float DeltaTime) {
-		AController* PlayerController = Cast<AController>(GetLocalFlowController()->GetOwner());
-		ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
+		bool bAllCharactersReachedDestination = true;
+		int32 Count = 0;
 
-		PlayerCharacter->AddMovementInput(FVector(1, 0, 0), 10.0f, true);
+		UGameInstance* GameInstance = GetGameInstance();
+		const FString WorkerId = GameInstance->GetSpatialWorkerId();
+		for (TActorIterator<ATestMovementCharacter> Iter(GetWorld()); Iter; ++Iter)
+		{
+			ATestMovementCharacter* Character = *Iter;
 
-		bCharacterReachedDestination =
-			GetTargetDistanceOnLine(Origin, Destination, PlayerCharacter->GetActorLocation()) > -20.0f; // 20cm overlap
+			if (Character->IsLocallyControlled())
+			{
+				Character->AddMovementInput(FVector(1.0f, 0.0f, 0.0f), 10.0f, true);
+				float PeakSpeed = Character->GetPeakSpeedInWindow();
+				AssertValue_Float(PeakSpeed, EComparisonMethod::Less_Than, 650.0f, TEXT("Actor not speeding"));
+			}
 
-		if (bCharacterReachedDestination)
-		{
-			AssertTrue(bCharacterReachedDestination, TEXT("Player character has reached the destination on the autonomous proxy."));
-			FinishStep();
+			bool bReachDestination = GetTargetDistanceOnLine(Origin, Destination, Character->GetActorLocation()) > -40.0f; // 20cm overlap
```

Comment:
```suggestion
			bool bReachDestination = GetTargetDistanceOnLine(Origin, Destination, Character->GetActorLocation()) > -40.0f; // 40cm overlap
```
? But also not actually sure how this is measured